client/assets/js/main.js

```diff
@@ -0,0 +1,236 @@
+const ZONES = ["Kök", "Säng", "Allrum nere", "Allrum uppe", "Toa nere", "Toa Uppe", "Bastu"]
+const SOURCES = ["iPad", "Computer", "Kitchen TV", "Upstairs TV"]
+const SEND_THROTTLE_MS = 100
+
+doubleTapCounter = 0
+
+let nextInputSendAllowed = 0
+let connectionAttempts = 0
+let nextConnectionAttempt = 0
+let webSocket
+
+const container = $("#container")
+
+showConnectingScreen()
+connectWebSocket()
+
+function connectWebSocket() {
+    webSocket = new WebSocket(`ws://${HOST}:8765`);
+    webSocket.onmessage = function (event) {
+        console.log(event.data);
+        jsn = JSON.parse(event.data)
+        if (jsn["responseType"] == "state") {
+            setState(jsn["data"]);
+        }
+    }
+
+    webSocket.onopen = function () {
+        connectionAttempts = 0
+        setupPage()
+    }
+
+    webSocket.onclose = function () {
+        if (connectionAttempts == 0) {
+            showConnectingScreen()
+        }
+        attemptReconnect()
+    }
+}
+
+function attemptReconnect() {
+    setTimeout(() => {
+        const nextReconnectionTime = Math.round((nextConnectionAttempt - now()) / 1000)
```

Comment:
Is now() from a library?

Reply:
It's defined in the very end of this file.
function now() {
    return (new Date).getTime();
}